lib/bolt/cli.rb

```diff
@@ -354,12 +354,18 @@ def validate(options)
                               "the project, run '#{command} #{options[:object]}'."
       end
 
-      if options[:subcommand] != 'file' && options[:subcommand] != 'script' &&
+      if !%w[file script lookup].include?(options[:subcommand]) &&
          !options[:leftovers].empty?
         raise Bolt::CLIError,
               "Unknown argument(s) #{options[:leftovers].join(', ')}"
       end
 
+      target_opts = options.keys.select { |opt| TARGETING_OPTIONS.include?(opt) }
+      if options[:subcommand] == 'lookup' &&
+         target_opts.any? && options[:plan_hierarchy]
+        raise Bolt::CLIError, "The 'lookup' command accepts either targetting option OR --plan-hierarchy."
```

Comment:
```suggestion
        raise Bolt::CLIError, "The 'lookup' command accepts either targeting option OR --plan-hierarchy."
```

I think listing out all options is reasonable here, just in case the user doesn't understand what we mean by 'targeting option'.